src/interfaces/node.cpp

```diff
@@ -152,10 +152,10 @@ class NodeImpl : public Node
     size_t getMempoolDynamicUsage() override { return ::mempool.DynamicMemoryUsage(); }
     bool getHeaderTip(int& height, int64_t& block_time) override
     {
-        LOCK(::cs_main);
-        if (::pindexBestHeader) {
-            height = ::pindexBestHeader->nHeight;
-            block_time = ::pindexBestHeader->GetBlockTime();
+        const CBlockIndex* best_header = ::GetBestHeader();
+        if (best_header) {
```

Comment:
Possible race condition here since we don't lock cs_main? Can the best_head data be deleted between time we call GetBestHeader() and when we use the ptr?

I'm not familiar with the code enough to know the answer to this so just putting it here in case.